### ISSUE TITLE
When uploading a File or Image: ValueError: I/O operation on closed file

### ISSUE THREAD

**Issue by carloshanson** [zopefoundation/Zope#152]
This is a new install of Zope 4.0a6 using virtualenv. Unfortunately, I am having trouble figuring out why. Running `./bin/runwsgi -v -d etc/zope.ini`, gives the following traceback when trying to upload an Image through the web:

```
2017-07-06 11:39:02 - ERROR [waitress] Exception when serving /operations-ticket/images/asset-tag/manage_upload
Traceback (most recent call last):
  File "/srv/zopepy2/local/lib/python2.7/site-packages/waitress/channel.py", line 338, in service
    task.service()
  File "/srv/zopepy2/local/lib/python2.7/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/srv/zopepy2/local/lib/python2.7/site-packages/waitress/task.py", line 399, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/httpexceptions.py", line 33, in __call__
    return self.application(environ, start_response)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/WSGIPublisher.py", line 238, in publish_module
    request, response, module_info, _publish=_publish)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/WSGIPublisher.py", line 211, in _publish_response
    reraise(exc.__class__, exc, sys.exc_info()[2])
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/WSGIPublisher.py", line 172, in _publish_response
    response = _publish(request, module_info)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/WSGIPublisher.py", line 162, in publish
    bind=1)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/mapply.py", line 85, in mapply
    return debug(object, args, context)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/ZPublisher/WSGIPublisher.py", line 57, in call_object
    return obj(*args)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/OFS/Image.py", line 506, in manage_upload
    data, size = self._read_data(file)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/OFS/Image.py", line 554, in _read_data
    seek(0, 2)
ValueError: I/O operation on closed file
```

When uploading a File, the traceback looks slightly different:
```
2017-07-06 11:44:34 - ERROR [waitress] Exception when serving /operations-ticket/images/manage_addProduct/OFSP/manage_addFile
Traceback (most recent call last):
[...]
  File "/srv/zopepy2/local/lib/python2.7/site-packages/OFS/Image.py", line 83, in manage_addFile
    newFile.manage_upload(file)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/OFS/Image.py", line 506, in manage_upload
    data, size = self._read_data(file)
  File "/srv/zopepy2/local/lib/python2.7/site-packages/OFS/Image.py", line 554, in _read_data
    seek(0, 2)
ValueError: I/O operation on closed file
```

I'm out of practice with Zope debugging. I will keep poking at it, but I would also appreciate pointers to become better and be more helpful.

Thanks.

**Comment by hannosch** [comment]
This is likely the same or a similar bug as #148.

**Comment by carloshanson** [comment]
After reviewing #148, I have to agree. The primary difference is the Python
version. I used Python 2.7.

Carlos Hanson

**Comment by carloshanson** [comment]
I'm going to close this ticket. I tested the change in the [comment ](https://github.com/zopefoundation/Zope/issues/148#issuecomment-313856974) by @stephan-hof. It also fixes the issue when using Python 2.7.
